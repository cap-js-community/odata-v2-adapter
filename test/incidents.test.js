"use strict";

const cds = require("@sap/cds");
const supertest = require("supertest");

const util = require("./_env/util/request");

cds.test(__dirname + "/_env");

let request;

describe("incidents", () => {
  beforeAll(async () => {
    await global._init;
    request = supertest(cds.app.server);
  });

  it("GET incidents service", async () => {
    const response = await util.callRead(request, "/odata/v2/incidents", {
      accept: "application/json",
    });
    expect(response.body).toBeDefined();
    expect(response.body.d.EntitySets.sort()).toMatchSnapshot();
  });

  it("GET incidents service data", async () => {
    const response = await util.callRead(request, "/odata/v2/incidents/Incidents", {
      accept: "application/json",
    });
    expect(response.body).toBeDefined();
    expect(response.body.d.results.length > 0).toEqual(true);
  });
});
