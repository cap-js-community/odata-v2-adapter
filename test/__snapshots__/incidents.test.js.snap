// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`incidents GET incidents service 1`] = `
[
  "Category",
  "Category_texts",
  "Country",
  "Country_texts",
  "Currency",
  "Currency_texts",
  "Incidents",
  "Priority",
  "Priority_texts",
]
`;
